### PR TITLE
Use Result.unwrap_or_else where applicable

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Review by lewiszlw — APPROVED** [review]
LGTM. Thanks @findepi .

File: datafusion/expr/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -1157,10 +1157,8 @@ impl LogicalPlan {
         Ok(if let LogicalPlan::Prepare(prepare_lp) = plan_with_values {
             param_values.verify(&prepare_lp.data_types)?;
             // try and take ownership of the input if is not shared, clone otherwise
-            match Arc::try_unwrap(prepare_lp.input) {
-                Ok(input) => input,
-                Err(arc_input) => arc_input.as_ref().clone(),
-            }
+            Arc::try_unwrap(prepare_lp.input)
+                .unwrap_or_else(|arc_input| arc_input.as_ref().clone())
```
Comment (jonahgao) [inline datafusion/expr/src/logical_plan/plan.rs:1161]: We can use [Arc::unwrap_or_clone](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.unwrap_or_clone) to simplify it.

File: datafusion/sql/src/unparser/rewrite.rs [hunk 2]
```diff
@@ -59,10 +59,8 @@ pub(super) fn normalize_union_schema(plan: &LogicalPlan) -> Result<LogicalPlan>
 
     let transformed_plan = plan.transform_up(|plan| match plan {
         LogicalPlan::Union(mut union) => {
-            let schema = match Arc::try_unwrap(union.schema) {
-                Ok(inner) => inner,
-                Err(schema) => (*schema).clone(),
-            };
+            let schema =
+                Arc::try_unwrap(union.schema).unwrap_or_else(|schema| (*schema).clone());
```
Comment (jonahgao) [inline datafusion/sql/src/unparser/rewrite.rs:63]: same as above

File: datafusion/expr/src/logical_plan/plan.rs (hunk 1 above)
Comment (findepi) [reply datafusion/expr/src/logical_plan/plan.rs]: good point, done!

**Comment by jonahgao** [comment]
Thanks @findepi @lewiszlw

**Comment by findepi** [comment]
thank you @lewiszlw @jonahgao !

**Pull request merged**